cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -454,29 +313,181 @@ angular.module(PKG.name + '.commons')
       };
     }
 
-    function formatConnections() {
-      var connections = [];
-      angular.forEach(vm.instance.getConnections(), function (conn) {
-        connections.push({
-          from: conn.sourceId,
-          to: conn.targetId
-        });
+    function addConnection(newConnObj) {
+      $scope.connections.push({
+        from: newConnObj.sourceId,
+        to: newConnObj.targetId
+      });
+      DAGPlusPlusNodesActionsFactory.setConnections($scope.connections);
+    }
+
+    function removeConnection(detachedConnObj, updateStore = true) {
+      var detachedConnection = _.find($scope.connections, function (conn) {
```

Comment:
nit: Lodash already has a findIndex function that we could use. If the findIndex doesn't return -1 then the connection exists. Could avoid doing finding and getting index and do it in one step.